messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -262,4 +262,44 @@ public MessageCheckpoint getMessageCheckpoint() {
   public MessageCheckpoint buildStartPartitionCheckpoints() {
     return null;
   }
+
+  @Override
+  public synchronized Long getReaderBackLog() throws IOException {
+    Long collectorStreamPendingSize = 0L, localStreamPendingSize = 0L;
+    int retryCount = 0;
+    int maxRetryThreshold = 10;
+    while (cReader.getCurrentFile() == null && retryCount < maxRetryThreshold) {
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        LOG.info("Sleep Interrupted while waiting for collector reader initialization. Ignoring the interrupt");
+      }
+      retryCount++;
+    }
+    //get collector reader remaining size
+    if (cReader.getCurrentFile() != null) {
+      collectorStreamPendingSize += cReader.getPendingSize(cReader.getCurrentFile());
+      LOG.info("Pending Size inside collector output location starting from collector file: " +
+              cReader.getCurrentFile() + " is " + collectorStreamPendingSize);
+    }
+    //get local reader remaining size
+    retryCount = 0;
+    while (lReader.getCurrentFile() == null && retryCount < maxRetryThreshold) {
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        LOG.info("Sleep Interrupted while waiting for local stream reader initialization. Ignoring the interrupt");
+      }
+      retryCount++;
+    }
+    if (lReader.getCurrentFile() != null) {
+      localStreamPendingSize += lReader.getPendingSize();
+      LOG.info("Pending Size inside streams_local output location starting from file: " +
+              lReader.getCurrentFile() + " is " + localStreamPendingSize);
+    }
+    LOG.info("Pending Size inside collector reader " + collectorStreamPendingSize
+            + " Pending Size inside local stream reader " + localStreamPendingSize
+            + " Total pending size " + (collectorStreamPendingSize + localStreamPendingSize));
+    return (collectorStreamPendingSize + localStreamPendingSize);
```

Comment:
I have examined around 100 files, from worker output, and found that after compression the files become 3.8 times smaller.  So I will add this factor while calculating the pending size